ext/hash/hash.c

```diff
@@ -1468,6 +1482,14 @@ PHP_METHOD(HashContext, __serialize)
 	Z_TRY_ADDREF(tmp);
 	zend_hash_next_index_insert(Z_ARRVAL_P(return_value), &tmp);
 
+	if (hash->args) {
+		ZVAL_ARR(&tmp, hash->args);
+		Z_TRY_ADDREF(tmp);
```

Comment:
Why TRY?

Reply:
Yeah, this should be always mutable, so no point.

Thanks.

Reply:
Why serialize the arguments at all? Leave it to `hash->ops->hash_serialize` to serialize arguments if necessary. I think it would be MUCH better for the hash arguments to not be stored independent of algorithm.